scripts/nb-tester/test-notebook.py

```diff
@@ -10,82 +10,166 @@
     nbformat.write(nb, filepath)
 
 
-def find_notebooks() -> list[Path]:
+def find_notebooks(*, submit_jobs: bool = False) -> list[Path]:
     """
     Get paths to all notebooks in NOTEBOOKS_GLOB that are not excluded by
     NOTEBOOKS_EXCLUDE
     """
     all_notebooks = Path(".").rglob(NOTEBOOKS_GLOB)
+    excluded_notebooks = NOTEBOOKS_EXCLUDE
+    if not submit_jobs:
+        excluded_notebooks += NOTEBOOKS_THAT_SUBMIT_JOBS
     return [
         path
         for path in all_notebooks
-        if not any(path.match(glob) for glob in NOTEBOOKS_EXCLUDE)
+        if not any(path.match(glob) for glob in excluded_notebooks)
     ]
 
 
+def cancel_trailing_jobs(start_time: datetime) -> bool:
+    """
+    Cancel any runtime jobs created after `start_time`.
+    Return True if non exist, False otherwise.
+
+    Notebooks should not submit jobs during a normal test run. If they do, the
+    cell will time out and this function will cancel the job to avoid wasting
+    device time.
+
+    If a notebook submits a job but does not wait for the result, this check
+    will also catch it and cancel the job.
+    """
+    service = QiskitRuntimeService()
+    jobs = [j for j in service.jobs(created_after=start_time) if not j.in_final_state()]
+    if not jobs:
+        return True
+
+    print(
+        f"⚠️ Cancelling {len(jobs)} job(s) created after {start_time}.\n"
+        "Add any notebooks that submit jobs to NOTEBOOKS_EXCLUDE in "
+        "`scripts/nb-tester/test-notebook.py`."
+    )
+    for job in jobs:
+        job.cancel()
+    return False
+
+
+def create_argument_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        prog="Notebook executor",
+        description="For testing notebooks and updating their outputs",
+    )
+    parser.add_argument(
+        "filenames",
+        help=(
+            "Paths to notebooks. If not provided, the script will search for "
+            "notebooks in `docs/`. To exclude a notebook from this process, add it "
+            "to `NOTEBOOKS_EXCLUDE` in the script."
+        ),
+        nargs="*",
+    )
+    parser.add_argument(
+        "-w",
+        "--write",
+        action="store_true",
+        help="Overwrite notebooks with the results of this script's execution.",
+    )
+    parser.add_argument(
+        "--submit-jobs",
+        action="store_true",
+        help=(
+            "Run notebooks that submit jobs to IBM Quantum and wait indefinitely "
+            "for jobs to complete. Warning: this has a real cost because it uses "
+            "quantum resources! Only use this argument occasionally and intentionally." 
+        ),
+    )
+    return parser
+
+
 if __name__ == "__main__":
-    args = sys.argv[1:]
-    write = WRITE_FLAG in args
-    if write:
-        args.remove(WRITE_FLAG)
+    args = create_argument_parser().parse_args()
 
-    notebook_paths = args or find_notebooks()
+    paths = map(Path, args.filenames or find_notebooks(args.submit_jobs))
```

Comment:
After adding the `*` in the function definition, we need to specify the argument here. Otherwise, it will fail when running the script without arguments.
```suggestion
    paths = map(Path, args.filenames or find_notebooks(submit_jobs=args.submit_jobs))
```

Reply:
Good spot, thanks!